### PR TITLE
download corrupted dir bug fix

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24811

## Description
Make sure to download successfully when dir is corrupted.

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/download/downloader.py [hunk 1]
```diff
@@ -478,7 +478,11 @@ def merge_and_extract_existing_file(self, custom_content_object: dict) -> None:
             searched_basename: str = self.get_searched_basename(file_name, ex_file_ending, ex_file_detail)
             corresponding_pack_file_object: dict = self.get_corresponding_pack_file_object(searched_basename,
                                                                                            corresponding_pack_object)
-            corresponding_pack_file_path: str = corresponding_pack_file_object['path']
+            if not corresponding_pack_file_object:
+                corresponding_pack_file_path: str = os.path.join(self.output_pack_path, file_entity,
+                                                                 self.create_dir_name(file_name), searched_basename)
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/download/downloader.py:483]: Can you explain what is happening here?

Comment (guyfreund) [reply demisto_sdk/commands/download/downloader.py:483]: If the file doesn't exist, we combine all the path parts and download the file to the path we've just created.

Comment (ShahafBenYakir) [reply demisto_sdk/commands/download/downloader.py:483]: Add an example comment on what is each path part. Will make the future debugging easier.
if an image doesn't exist how can you create one?

Comment (guyfreund) [reply demisto_sdk/commands/download/downloader.py:483]: I download the unified intergation, extract it and then I have the image, I just have to create the path to download the image to.

File: demisto_sdk/commands/download/tests/downloader_test.py [hunk 2]
```diff
@@ -339,6 +339,33 @@ def test_update_pack_hierarchy(self, tmp_path):
...
+            ryaml = YAML()
+            ryaml.preserve_quotes = True
+            downloader.output_pack_path = env.PACK_INSTANCE_PATH
+            downloader.log_verbose = False
+            downloader.pack_content = env.PACK_CONTENT
+            downloader.run_format = False
+            downloader.num_merged_files = 0
+            downloader.num_added_files = 0
+            downloader.log_verbose = False
+            downloader.merge_and_extract_existing_file(env.INTEGRATION_CUSTOM_CONTENT_OBJECT)
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/download/tests/downloader_test.py:367]: Are we intentionally not asserting anything? You can always assert the status code.

**Pull request merged**